crates/wasmtime/src/module.rs

```diff
@@ -268,21 +229,20 @@ impl Module {
     /// # }
     /// ```
     pub fn from_file(engine: &Engine, file: impl AsRef<Path>) -> Result<Module> {
-        #[cfg(feature = "wat")]
-        let wasm = wat::parse_file(file)?;
-        #[cfg(not(feature = "wat"))]
-        let wasm = std::fs::read(file)?;
-        Module::new(engine, &wasm)
+        Self::new(
+            engine,
+            &fs::read(file).with_context(|| "failed to read input file")?,
```

Comment:
FWIW using `wat::parse_file` was done before because it automatically gives better error messages if the input file is `*.wat`. We can give rustc-style errors with filenames/line numbers and snippets outlined.

Reply:
Oh whoops, I missed the `set_path` call in the error handling.  I'll change this back.

Reply:
As I'd like to avoid opening the file twice (once to check if the AOT header is present before handing it off to `wat::parse_file` which opens the file in turn), do you think we could add a `wat::Error::set_path` method that sets the path if the underlying error kind is `Wast`?

Reply:
I've opened https://github.com/bytecodealliance/wasm-tools/pull/252 that would allow us to set the path in the error without having to resort to opening the file twice.